types.ts

```diff
@@ -1,11 +1,17 @@
 import { NextPage } from "next";
-import { AppProps } from "next/app";
-import { ReactNode } from "react";
+import { AppProps as NextAppProps } from "next/app";
+import { NextComponentType, NextPageContext } from "next/types";
+import { ReactElement, ReactNode } from "react";
 
-export type PageWithLayout<P = {}> = NextPage<P> & {
-  getLayout?: (page: ReactNode) => ReactNode;
+export type PageWithLayout = NextPage & {
+  getLayout?: (page: ReactElement) => ReactNode;
```

Comment:
I'm not a React expert, but it looks `ReactNode` is the more correct one for `page` attribute, as:
> A ReactNode is a ReactElement, a ReactFragment, a string, a number or an array of ReactNodes, or null, or undefined, or a boolean:

(see: https://stackoverflow.com/a/58123882)

At the same moment these are types and the app is building right now.

Reply:
look here https://nextjs.org/docs/basic-features/layouts

Reply:
@magul Element is more narrow, page can't be null or string, so I think Element is better?